src/files.ts

```diff
@@ -79,8 +82,11 @@ export function getFileMethods(
                 const id = this.file_id;
                 throw new Error(`File path is not available for file '${id}'`);
             }
-            if (isAbsolutePath(path)) return path;
-            const link = linkBuilder(path);
+            if (isAbsolutePath(path)) {
+                var bpath = pathBuilder(path);
+                if (bpath === undefined) return path;
+            }
+            const link = bpath ?? linkBuilder(path);
```

Comment:
I'm not sure what your code was supposed to do, I expected something like this?
```suggestion
            if (isAbsolutePath(path)) return pathBuilder(path);
            const link = linkBuilder(path);
```

Reply:
`pathBuilder` can't be called if the user intentionally wants to use an absolute path (aka local copy instead of download). 

If it's an absolute path, check whether `pathBuilder` is working. If it is, use what it returned that as `path`, if not, return `path` so it can copy the file rather than download as it's a local file. 
If it's not an absolute path, use `linkBuilder`.

But still, you're right. Simple solution. I'll make `pathBuilder` return `path` if not defined by user.